storage/storage.go

```diff
@@ -19,15 +19,11 @@ type Storage interface {
 	AddPiece(ctx context.Context, sector abi.SectorID, pieceSizes []abi.UnpaddedPieceSize, newPieceSize abi.UnpaddedPieceSize, pieceData Data) (abi.PieceInfo, error)
 }
 
-type FallbackPostOut struct {
-	PoStInputs []PoStCandidateWithTicket
-	Proof      []abi.PoStProof
-}
-
 type Prover interface {
-	GenerateEPostCandidates(ctx context.Context, miner abi.ActorID, sectorInfo []abi.SectorInfo, challengeSeed abi.PoStRandomness, faults []abi.SectorNumber) ([]PoStCandidateWithTicket, error)
-	GenerateFallbackPoSt(ctx context.Context, miner abi.ActorID, sectorInfo []abi.SectorInfo, challengeSeed abi.PoStRandomness, faults []abi.SectorNumber) (FallbackPostOut, error)
-	ComputeElectionPoSt(ctx context.Context, miner abi.ActorID, sectorInfo []abi.SectorInfo, challengeSeed abi.PoStRandomness, winners []abi.PoStCandidate) ([]abi.PoStProof, error)
+	GenerateWinningPoStSectorChallenge(proofType abi.RegisteredProof, minerID abi.ActorID, randomness abi.PoStRandomness, eligibleSectorCount uint64) ([]uint64, error)
+	GenerateWinningPoSt(minerID abi.ActorID, sectorInfo []abi.SectorInfo, randomness abi.PoStRandomness) ([]abi.PoStProof, error)
+
+	GenerateWindowPoSt(minerID abi.ActorID, privateSectorInfo []abi.SectorInfo, randomness abi.PoStRandomness) ([]abi.PoStProof, error)
```

Comment:
Nitpick: `GenerateWinningPoSt` and `GenerateWindowPoSt` each have a parameter of type `abi.SectorInfo`. I suggest calling both parameters `sectorInfo`.